File: 3629-total-characters-in-string-after-transformations-i/3629-total-characters-in-string-after-transformations-i.cpp
class Solution {
public:
    int lengthAfterTransformations(string s, int t) {
        const int MOD = 1e9+7;
        vector<long long> freq(26,0);
        for(char c: s) freq[c-'a']++;
        while(t){
            vector<long long> tempfreq(26,0);
            for(int i = 0; i<26; i++){
                if(freq[i]){
                    if(i== 25){
                        tempfreq[0] = (tempfreq[0] + freq[i])%MOD;
                        tempfreq[1] = (tempfreq[1] + freq[i])%MOD;
                    }
                    else tempfreq[i+1] = (tempfreq[i+1] + freq[i])%MOD;
                }
            }
            freq = tempfreq;
            t--;
        }
        int ans = 0;
        for(int val : freq) ans = (ans+val)%MOD;
        return ans;
    }
};

File: 3629-total-characters-in-string-after-transformations-i/3629-total-characters-in-string-after-transformations-i.java
class Solution {
    public int lengthAfterTransformations(String s, int t) {
        final int MOD = 1000000007;
        long[] freq = new long[26];
        for(char c : s.toCharArray()) freq[c-'a']++;
        while(t>0){
            long[] tempFreq = new long[26];
            for(int i = 0; i<26; i++){
                if(freq[i]>0){
                    if(i == 25){
                        tempFreq[0] = (freq[i] + tempFreq[0])%MOD;
                        tempFreq[1] = (freq[i] + tempFreq[1])%MOD;
                    }else tempFreq[i+1] = (freq[i] + tempFreq[i+1])%MOD;
                }
            }
            freq = tempFreq;
            t--;
        }
        int ans = 0;
        for(long val : freq) ans = (ans + (int)val)%MOD;
        return ans;
    }
}